tests/cpgrid/adapt_cpgrid_test.cpp

```diff
@@ -370,6 +372,14 @@ void markAndAdapt_check(Dune::CpGrid& coarse_grid,
             } // end-preAdaptElements-for-loop
         } // end-startingGridIdx==0
 
+        for (int level = 1; level < coarse_grid.maxLevel(); ++level)
+        {
+            auto itMinLevel = std::min_element((data[level] -> global_cell_).begin(),  (data[level] -> global_cell_).end());
+            auto itMaxLevel = std::max_element((data[level] -> global_cell_).begin(),  (data[level] -> global_cell_).end());
+            BOOST_CHECK_EQUAL( *itMinLevel, 0);
+            BOOST_CHECK_EQUAL( *itMaxLevel, data[level]-> size(0) -1);
```

Comment:
Same here and below.  There could be inactive parent cells in the region to refine and hence missing kids (in the future?).

Reply:
Now the check is done with inequalities, meaning that the maximum value of global cell should be between 0 and the maximum Cartesian Index (the later might vary for each LGR, and it's equal to the maximum Cartesian Index from level zero for the leaf grid view)